app/controllers/prison/visits_controller.rb

```diff
@@ -4,15 +4,11 @@ class Prison::VisitsController < ApplicationController
   before_action :require_login_during_trial,
     only: %w[show nomis_cancelled process_visit update]
   before_action :cancellation_reason_set, only: :cancel
+  before_action :visit_is_processable, only: [:process_visit, :update]
```

Comment:
I think having this as a before action is slightly clearer that both require the visit is processable as well as making the action methods more concise. We load the visit twice, but as you say they are memoised so I see no problem.